server/controllers/profile.js

```diff
@@ -0,0 +1,194 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile/create
+// @desc Create New Profile
+// @desc Private
+
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    const DAYS = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
+
+    const user = await User.findById(req.user.id);
+
+    if (!user) {
+        res.status(401)
+        throw new Error("Not authorized")
+    }
+
+    const profileExists = await Profile.findOne({ user })
+
+    if (profileExists) {
+        res.status(400);
+        throw new Error("A profile has already been created for this user")
+    }
+
+    const {first_name, last_name, description, availability, gender, birth_date, phone_numbers, address} = req.body;
+
+    //Create an array of days, removing any invalid days
+    const availabilityArray = availability.split(',').filter((day)=> DAYS.includes(day))
+
+    //Create an array of phone numbers
+    const phoneNumberArray = phone_numbers.split(',')
+
+    const newProfile = await Profile.create({
+        first_name, last_name, description, availability: availabilityArray, gender, birth_date, email_address: user.email, phone_numbers: phoneNumberArray, address, user
+    })
+
+    if (newProfile){
+        res.status(201).json({
+            success: {
+                profile: {
+                    first_name: newProfile.first_name, 
+                    last_name: newProfile.last_name, 
+                    description: newProfile.description,
+                    availability: newProfile.availability,
+                    gender: newProfile.gender,
+                    birth_date: newProfile.birth_date,
+                    email_address: newProfile.email_address,
+                    phone_numbers: newProfile.phone_numbers,
+                    address: newProfile.address,
+                    user: newProfile.user
+                }
+            }
+        })
+    } else {
+        res.status(400);

```

Comment:
```suggestion
        res.status(500);
```
This will be a server error here

Reply:
resolved by 143a61580e40df0a00e4f127e7f5e1859611a2b3

Reply:
I did not commit your change because I formatted it with Prettier, which made offset the diffs. In the comment above, I have provided the corresponding commit that fixes this issue.